### PR TITLE
added disclosure widget issue_48

### LINKED ISSUE #48: Add any exciting feature of your own which has a usecase in project

### PR DESCRIPTION
**Pull request by disvid**
### Related Issue  
Closes #48 
### Type of Change
Put `x` inside the square bracket to specify what type of change your PR is:  
- [x] New Feature  
- [ ] Bug Fix  
- [ ] Code Refactor  
- [ ] Documentation Update  
- [ ] Other (please specify): 

### Description of Change  
Disclosure widget feature has been created.

### Demo  

[Screencast from 2024-12-30 23-23-31.webm](https://github.com/user-attachments/assets/a262df34-550f-47ff-ac6a-d61499b6231c)

### CODE REVIEW

File: app/page.jsx [hunk 1]
```diff
@@ -105,23 +114,33 @@ const Dashboard = () => {
...
-                        ticket={filteredTicket}
-                      />
-                    ))}
+                <div
+                  className="flex justify-between items-center cursor-pointer bg-black-200 p-2 rounded"
+                  onClick={() => toggleGroupVisibility(group)}
+                >
+                  <h2 className="font-bold">{group}</h2>
+                  <span>
+                    {groupVisibility[group] ? "▼" : "►"}
```
Comment (AyushDharDubey) [inline app/page.jsx:123]: please avoid using the symbols directly.
```javascript
{groupVisibility[group] ? "▼" : "►"}
```

instead, please explore suitable FontAwesome icons for better consistency.

**Comment by AyushDharDubey** [comment]
also, it would be better if we could display 3-4 tickets initially and use a disclosure widget to control the visibility of the remaining tickets.

**Comment by disvid** [comment]
Done @AyushDharDubey

**Comment by AyushDharDubey** [comment]
did you test the change on your local before pushing?

**Comment by disvid** [comment]
yes sir @AyushDharDubey

**Comment by disvid** [comment]
@AyushDharDubey just uploaded the screenrecording on PR description

**Comment by AyushDharDubey** [comment]
ok good work @disvid,  
i've also confirmed on my end that the fix is indeed working.  

however, there is a slight issue. you've updated the packages, which has caused backward incompatibility in the project. This **will** create problems for contributors already working on the project unless they run `npm i` on their end. to avoid wreaking havoc on everybody's system, please change

```javascript
<FontAwesomeIcon
  icon={groupVisibility[group] ? faChevronDown : faChevronRight}
/>
```  
to:  
```javascript
<FontAwesomeIcon
  icon={groupVisibility[group] ? faChevronDown : faChevronRight}
  className="icon"
/>
```  

that would be all, thanks for your efforts
i will be waiting to merge this one

**Comment by disvid** [comment]
@AyushDharDubey Done sir,made the change.

**Pull request merged**